pathway-analysis/02-gene_set_enrichment_analysis.Rmd

```diff
@@ -111,77 +107,143 @@ dge_results_df <- readr::read_tsv(dge_results_file)
 head(dge_results_df)
 ```
 
-We can take the same steps we did earlier to convert these Ensembl gene IDs to Entrez IDs. 
-We have to change the `column` argument to `mapIds()` (and the downstream steps).
+Since this data frame of DGE results includes gene symbols, we do not need to perform any kind of gene identifier conversion.
+We do, however, need to check for duplicate gene symbols. 
+We can accomplish this with `duplicated()`, which returns a logical vector (e.g., `TRUE` or `FALSE`).
+The function `any()` will return `TRUE` if **any** value it is given `TRUE`.
+
+```{r any_duplicated, live = TRUE}
+any(duplicated(dge_results_df$gene_symbol))
+```
+
+This will cause a problem when we go to run GSEA.
+Let's talk more about the method itself to understand _why_.
+
+## Gene Set Enrichment Analysis
```

Comment:
I think this explanation should come before the duplication discovery. Placing it here makes it seem like the only reason you are explaining is to understand why duplicates cause trouble, but I think it is worth more than that! (I am sure you don't disagree.)

I think I would move this explanation of GSEA right to the top, followed by the gene sets, then by the duplicate removal. 

So my organization might look like this:
- Gene Set Enrichment Analysis Introduction
  - Gene sets
- Differential gene expression results
   - Removing Duplicates
   -  Pre-ranked list
-  Run GSEA